core/kernel/tee_ta_manager.c

```diff
@@ -481,6 +534,9 @@ TEE_Result tee_ta_open_session(TEE_ErrorOrigin *err,
 		return res;
 	}
 
+	if (!check_params(s, param))
+		return TEE_ERROR_BAD_PARAMETERS; /* intentional generic error */
```

Comment:
I don't get the comment. What is generic here? The error code looks very specific to me, it means "bad parameters", how could you be more specific anyway? ;)
Same below.